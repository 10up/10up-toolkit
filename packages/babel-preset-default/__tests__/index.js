const path = require('path');
const fs = require('fs');
const babel = require('@babel/core');
const babelPresetDefault = require('../index');

// const nodeModulesPath = path.join(__dirname, '../node_modules');
const reactPath = path.dirname(require.resolve('react'));

const es6Input = fs.readFileSync(path.join(__dirname, '../fixtures/es6+.js'));
const reactInput = fs.readFileSync(path.join(__dirname, '../fixtures/react.js'));
const tsInput = fs.readFileSync(path.join(__dirname, '../fixtures/typescript.ts'));

describe('Babel preset default', () => {
	beforeEach(() => {
		if (fs.existsSync(path.join(reactPath, 'jsx-runtime.js'))) {
			fs.renameSync(
				path.join(reactPath, 'jsx-runtime.js'),
				path.join(reactPath, 'jsx-runtime2.js'),
			);
		}
	});

	afterEach(() => {
		if (fs.existsSync(path.join(reactPath, 'jsx-runtime2.js'))) {
			fs.renameSync(
				path.join(reactPath, 'jsx-runtime2.js'),
				path.join(reactPath, 'jsx-runtime.js'),
			);
		}
	});
	test('transpiles ES6+ code properly', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles ES6+ code properly with runtimeESModules set to true', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { runtimeESModules: true }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles react code properly', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles wordpress code properly', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { wordpress: true }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	// Babel will automatically set modules to false when called by Webpack.
	test('transpiles without transforming es6 imports', () => {
		const output = babel.transform(es6Input, {
			configFile: false,
			envName: 'production',
			presets: [[babelPresetDefault, { modules: false }]],
		});

		expect(output.code).toMatchSnapshot();
	});

	test('transpiles without removing proptypes in dev mode', () => {
		const output = babel.transform(reactInput, {
			configFile: false,
			envName: 'development',
			presets: [[babelPresetDefault]],
		});

		expect(output.code).toMatchSnapshot();
	});
});

describe('compiles typescript', () => {
	test('transpiles typescript code properly', () => {
		const output = babel.transform(tsInput, {
			configFile: false,
			envName: 'production',
			presets: [babelPresetDefault],
		});

		expect(output.code).toMatchSnapshot();
	});
});
