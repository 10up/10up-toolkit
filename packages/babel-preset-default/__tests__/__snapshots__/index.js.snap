// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel preset default transpiles ES6+ code properly 1`] = `
"\\"use strict\\";

const constVar = 'someConstVar';
const promise = new Promise(resolve => {
  setTimeout(() => {
    resolve(\`Resolving \${constVar}\`);
  }, 100);
});

function* generator() {
  yield 1;
  yield 2;
  yield 3;
}

const object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

const init = async () => {
  const result = await promise();

  for (const a of generator()) {
    console.log(a);
  }

  console.log(object?.a?.b);
  return result;
};

init().then(result => {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles ES6+ code properly with runtimeESModules set to true 1`] = `
"\\"use strict\\";

const constVar = 'someConstVar';
const promise = new Promise(resolve => {
  setTimeout(() => {
    resolve(\`Resolving \${constVar}\`);
  }, 100);
});

function* generator() {
  yield 1;
  yield 2;
  yield 3;
}

const object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

const init = async () => {
  const result = await promise();

  for (const a of generator()) {
    console.log(a);
  }

  console.log(object?.a?.b);
  return result;
};

init().then(result => {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles react code properly 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = require(\\"react\\");

// eslint-disable-line import/no-extraneous-dependencies
const FCComponent = ({
  title,
  url,
  description
}) => /*#__PURE__*/React.createElement(\\"div\\", {
  className: \\"component-test-babel\\"
}, /*#__PURE__*/React.createElement(\\"a\\", {
  href: url,
  title: title
}, /*#__PURE__*/React.createElement(\\"h1\\", null, title)), /*#__PURE__*/React.createElement(\\"p\\", null, description));

FCComponent.defaultProps = {
  title: '',
  url: '',
  description: ''
};

class ClassComponent extends _react.Component {
  componentDidMount() {// doing something
  }

  render() {
    const {
      title
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"Class Component\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, title));
  }

}

var _default = FCComponent;
exports.default = _default;"
`;

exports[`Babel preset default transpiles without removing proptypes in dev mode 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = require(\\"react\\");

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

var _jsxFileName = \\"\\";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// eslint-disable-line import/no-extraneous-dependencies
const FCComponent = ({
  title,
  url,
  description
}) => /*#__PURE__*/React.createElement(\\"div\\", {
  className: \\"component-test-babel\\",
  __self: void 0,
  __source: {
    fileName: _jsxFileName,
    lineNumber: 5,
    columnNumber: 2
  }
}, /*#__PURE__*/React.createElement(\\"a\\", {
  href: url,
  title: title,
  __self: void 0,
  __source: {
    fileName: _jsxFileName,
    lineNumber: 6,
    columnNumber: 3
  }
}, /*#__PURE__*/React.createElement(\\"h1\\", {
  __self: void 0,
  __source: {
    fileName: _jsxFileName,
    lineNumber: 7,
    columnNumber: 4
  }
}, title)), /*#__PURE__*/React.createElement(\\"p\\", {
  __self: void 0,
  __source: {
    fileName: _jsxFileName,
    lineNumber: 9,
    columnNumber: 3
  }
}, description));

FCComponent.propTypes = {
  title: _propTypes.default.string,
  url: _propTypes.default.string,
  description: _propTypes.default.string
};
FCComponent.defaultProps = {
  title: '',
  url: '',
  description: ''
};

class ClassComponent extends _react.Component {
  componentDidMount() {// doing something
  }

  render() {
    const {
      title
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"Class Component\\",
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 33,
        columnNumber: 4
      }
    }, /*#__PURE__*/React.createElement(\\"h1\\", {
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 34,
        columnNumber: 5
      }
    }, title));
  }

}

ClassComponent.propTypes = {
  title: _propTypes.default.string.isRequired
};
var _default = FCComponent;
exports.default = _default;"
`;

exports[`Babel preset default transpiles without transforming es6 imports 1`] = `
"const constVar = 'someConstVar';
const promise = new Promise(resolve => {
  setTimeout(() => {
    resolve(\`Resolving \${constVar}\`);
  }, 100);
});

function* generator() {
  yield 1;
  yield 2;
  yield 3;
}

const object = {
  a: {
    b: 'b'
  },
  b: {
    a: 'a'
  }
};

const init = async () => {
  const result = await promise();

  for (const a of generator()) {
    console.log(a);
  }

  console.log(object?.a?.b);
  return result;
};

init().then(result => {
  console.log(result);
});"
`;

exports[`Babel preset default transpiles wordpress code properly 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _element = require(\\"@wordpress/element\\");

var _react = require(\\"react\\");

// eslint-disable-line import/no-extraneous-dependencies
const FCComponent = ({
  title,
  url,
  description
}) => (0, _element.createElement)(\\"div\\", {
  className: \\"component-test-babel\\"
}, (0, _element.createElement)(\\"a\\", {
  href: url,
  title: title
}, (0, _element.createElement)(\\"h1\\", null, title)), (0, _element.createElement)(\\"p\\", null, description));

FCComponent.defaultProps = {
  title: '',
  url: '',
  description: ''
};

class ClassComponent extends _react.Component {
  componentDidMount() {// doing something
  }

  render() {
    const {
      title
    } = this.props;
    return (0, _element.createElement)(\\"div\\", {
      className: \\"Class Component\\"
    }, (0, _element.createElement)(\\"h1\\", null, title));
  }

}

var _default = FCComponent;
exports.default = _default;"
`;

exports[`Babel preset default with automatic runtime transpiles react code properly with automatic runtime 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = require(\\"react\\");

// eslint-disable-line import/no-extraneous-dependencies
const FCComponent = ({
  title,
  url,
  description
}) => /*#__PURE__*/React.createElement(\\"div\\", {
  className: \\"component-test-babel\\"
}, /*#__PURE__*/React.createElement(\\"a\\", {
  href: url,
  title: title
}, /*#__PURE__*/React.createElement(\\"h1\\", null, title)), /*#__PURE__*/React.createElement(\\"p\\", null, description));

FCComponent.defaultProps = {
  title: '',
  url: '',
  description: ''
};

class ClassComponent extends _react.Component {
  componentDidMount() {// doing something
  }

  render() {
    const {
      title
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"Class Component\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, title));
  }

}

var _default = FCComponent;
exports.default = _default;"
`;

exports[`Babel preset default with automatic runtime transpiles wordpress code properly even when jsx-runtime.js is present 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _element = require(\\"@wordpress/element\\");

var _react = require(\\"react\\");

// eslint-disable-line import/no-extraneous-dependencies
const FCComponent = ({
  title,
  url,
  description
}) => (0, _element.createElement)(\\"div\\", {
  className: \\"component-test-babel\\"
}, (0, _element.createElement)(\\"a\\", {
  href: url,
  title: title
}, (0, _element.createElement)(\\"h1\\", null, title)), (0, _element.createElement)(\\"p\\", null, description));

FCComponent.defaultProps = {
  title: '',
  url: '',
  description: ''
};

class ClassComponent extends _react.Component {
  componentDidMount() {// doing something
  }

  render() {
    const {
      title
    } = this.props;
    return (0, _element.createElement)(\\"div\\", {
      className: \\"Class Component\\"
    }, (0, _element.createElement)(\\"h1\\", null, title));
  }

}

var _default = FCComponent;
exports.default = _default;"
`;

exports[`compiles typescript transpiles typescript code properly 1`] = `
"\\"use strict\\";

function helloTypeScript(name) {
  return \`Hello \${name}, this is TypeScript!\`;
}"
`;
